### PR TITLE
fs: add more unix-only fields to FileInfo

### LINKED ISSUE #3668: `FileInfo` should include same stats as <sys/stat.h>
[Node](https://nodejs.org/api/fs.html#fs_class_fs_stats) and [QuickJS](https://bellard.org/quickjs/quickjs.html#os-module) both return an object representing a standard C [`stat(2)`](http://man7.org/linux/man-pages/man2/stat.2.html) `struct`:

~~~console
λ Desktop: qjs
QuickJS - Type "\h" for help
qjs > os.lstat(".")[0]
{ dev: 16777220, ino: 13535700, mode: 16832, nlink: 10, uid: 501, gid: 20, rdev: 0, size: 340, blocks: 0, atime: 1578914527000, mtime: 1578912489000, ctime: 1578914526000 }

λ Desktop: node
Welcome to Node.js v13.6.0.
Type ".help" for more information.
> fs.statSync(".")
Stats {
  dev: 16777220,
  mode: 16832,
  nlink: 10,
  uid: 501,
  gid: 20,
  rdev: 0,
  blksize: 4096,
  ino: 13535700,
  size: 340,
  blocks: 0,
  ︙
}
~~~

Deno, however, [does not](https://deno.land/typedoc/interfaces/deno.fileinfo.html#mode):

~~~console
λ Desktop: deno
> Deno.statSync(".")
FileInfoImpl { _res, _isFile, _isSymlink, len, modified, accessed, created, mode, name }
~~~

It'd be nice if Deno returned the same set of information. Also, I recommend renaming `len` to `size`, because the former name is rather confusing (not to mention inconsistent with other implementations).

### PR DESCRIPTION
**Pull request by kevinkassimo**
Closes #3668

TODO:
+ ~~Decide whether we should go with these libc names, or renaming them~~
+ ~~Decide whether we should add `atime_nsec` and others in this PR (require `BigInt` repr)~~
+ [x] Add tests

### CODE REVIEW

File: cli/js/stat.ts [hunk 1]
```diff
@@ -10,9 +10,18 @@ export interface StatResponse {
   modified: number;
   accessed: number;
   created: number;
-  mode: number;
-  hasMode: boolean; // false on windows
   name: string | null;
+  isUnix: boolean; // false on windows
```
Comment (ry) [inline cli/js/stat.ts]: Why isUnix? Is that in Node?

Comment (ry) [reply cli/js/stat.ts]: But it seems unnecessary - we have ways of checking if windows without including this: `Deno.build.os`

Comment (kevinkassimo) [reply cli/js/stat.ts]: Makes sense.

**Comment by kevinkassimo** [comment]
Yeah there should be no extra syscall (should be cached)

**Comment by ry** [comment]
> Decide whether we should go with these libc names, or renaming them

Probably better to match node names. Which I think you’re already doing?

> Decide whether we should add atime_nsec and others in this PR (require BigInt repr)

Not necessary as far as I’m concerned. I’d leave it until someone wants it.

**Comment by kevinkassimo** [comment]
@ry Node names are mostly libc names so yeah they should match.

**Pull request merged**